.azure-pipelines.yml

```diff
@@ -0,0 +1,248 @@
+        rustup_toolchain: stable
+        build_type: release
+      mac-stable-release:
+        imageName: 'macos-10.14'
+        rustup_toolchain: stable
+        build_type: release
+      linux-stable:
+        imageName: 'ubuntu-16.04'
+        rustup_toolchain: stable
+      mac-stable:
+        imageName: 'macos-10.14'
+        rustup_toolchain: stable
+      mac-beta:
+        imageName: 'macos-10.14'
+        rustup_toolchain: beta
+      mac-nightly:
+        imageName: 'macos-10.14'
+        rustup_toolchain: nightly
+
+  pool:
+    vmImage: $(imageName)
+
+  variables:
+    SCCACHE_DIR: $(Pipeline.Workspace)/.sccache
+    RUSTC_WRAPPER: sccache
+
+  steps:
+    - checkout: self
+      submodules: true
+    - script: |
+        curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain $RUSTUP_TOOLCHAIN
+        echo "##vso[task.setvariable variable=PATH;]$PATH:$HOME/.cargo/bin"
+        export PATH=$PATH:$HOME/.cargo/bin
+        rustup update $RUSTUP_TOOLCHAIN
+        rustup default $RUSTUP_TOOLCHAIN
+        rustc --version
+      displayName: Install rust (*nix)
+      condition: ne( variables['Agent.OS'], 'Windows_NT' )
+    - script: |
+        rustup update %RUSTUP_TOOLCHAIN%
+        rustup default %RUSTUP_TOOLCHAIN%
+        rustc --version
+      displayName: Install rust (Win)
+      condition: eq( variables['Agent.OS'], 'Windows_NT' )
+    - script: |
+        rustup component add rustfmt
+      displayName: Add rustfmt
+      condition: and(eq(variables['rustup_toolchain'], 'stable'), ne( variables['build_type'], 'release' ),
+                     eq(variables['Agent.OS'], 'Darwin'))
+    - script: |
+        powershell -Command "$ProgressPreference = 'SilentlyContinue'; iwr -outf %TEMP%\LLVM-8.0.0-win64.exe https://rust-lang-ci2.s3.amazonaws.com/rust-ci-mirror/LLVM-8.0.0-win64.exe"
+        %TEMP%\LLVM-8.0.0-win64.exe /S /NCRC
+      condition: eq(variables['Agent.OS'], 'Windows_NT')
+      displayName: Install clang (Windows)
+    - bash: echo "##vso[task.setvariable variable=RUSTC_VERSION;]`rustc --version`"
+      displayName: Set rustc version string for caching
+    - bash: |
+        echo "##vso[task.setvariable variable=tagName;]`echo $BUILD_SOURCEBRANCH | sed -e 's|refs/tags/||'`"
+      displayName: Set tag name
+      condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/')
+    - bash: |
+        echo "##vso[task.setvariable variable=tagName;]latest-master"
+      displayName: Set tag name to "latest-master"
+      condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
+
+    - bash: |
+        set -e
+        curl -fo /usr/local/bin/sccache https://rust-lang-ci2.s3.amazonaws.com/rust-ci-mirror/2018-04-02-sccache-x86_64-apple-darwin
+        chmod +x /usr/local/bin/sccache
+      displayName: Install sccache (OSX)
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+    - bash: |
+        set -e
+        sudo apt-get install sccache -y
```

Comment:
FWIW I'd recommend using the same `sccache` version across all the platforms to ensure consistency.

I also personally like to extract these sorts of isolated steps to [separate files](https://github.com/rustwasm/wasm-bindgen/blob/master/ci/azure-install-sccache.yml) which may work well here too?

(it's probably also better to use the official sccache release binaries instead of the custom ones for rust-lang/rust)

Reply:
I changed this to use the [official sscache releases](https://github.com/mozilla/sccache/releases) from github. That sort of works, except [Windows breaks in all kinds of interesting ways](https://dev.azure.com/till0325/Wasmtime/_build/results?buildId=133) now, and for some reason the cache doesn't actually work. (See the `sccache` and `pre-compile sccache stats` tasks in [this build](https://dev.azure.com/till0325/Wasmtime/_build/results?buildId=132): the cache is restored, but sccache thinks it's empty.)

I disabled the entire cache for now: runs take 15 minutes, with Windows release builds being the long pole, and that seems acceptable for the time being.